waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -211,6 +214,13 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
                 ClusterManager clusterManager = new ClusterManagerImpl(zkClient, zkRoot, partitionAssignmentPolicy);
                 int numPartitions = clusterManager.numPartitions();
 
+                if (cmd.hasOption("partition")) {
+                    pId = Integer.parseInt(cmd.getOptionValue("partition"));
+                    if ((pId < 0) || (pId >= numPartitions)) {
+                        throw new IllegalArgumentException("Partition " + pId + " is not valid.");
+                    }
+                }
+
                 for (int partitionId = 0; partitionId < clusterManager.numPartitions(); partitionId++) {
```

Comment:
naming nit: It is better to change this for-local variable to pId and use partitionId for the variable with larger scope.